dbos/_sys_db.py

```diff
@@ -314,6 +320,8 @@ def insert_workflow_status(
                 recovery_attempts=(
                     1 if wf_status != WorkflowStatusString.ENQUEUED.value else 0
                 ),
+                workflow_timeout_ms=status["workflow_timeout_ms"],
+                workflow_deadline_epoch_ms=status["workflow_deadline_epoch_ms"],
             )
             .on_conflict_do_update(
```

Comment:
For queued workflows, when it gets started, should we also update/record the new workflow_deadline? Then we need on conflict update `workflow_deadline_epoch_ms`

Reply:
For queued workflows, status["workflow_deadline_epoch_ms"] is always `None`. This is all explained in `_get_timeout_deadline`, which is where these inputs come from. `insert_workflow_status` is a deliberately "dumb" function that just inserts what's given to it, I don't want any complex logic here.

Reply:
But how does recovery work? If a queued workflow interrupted, then the deadline is not set to the correct one when it recovers. Did I miss something here?

Reply:
The deadline for a queued workflow is set on dequeue. After that, it works just like a normal workflow.

Reply:
Uh missed the code branch for start queued workflow.